src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -299,53 +340,93 @@ public static void RemoveZip64Blocks(List<ZipGenericExtraField> extraFields)
 
         public void WriteBlock(Stream stream)
         {
-            BinaryWriter writer = new BinaryWriter(stream);
-            writer.Write(TagConstant);
-            writer.Write(_size);
-            if (_uncompressedSize != null) writer.Write(_uncompressedSize.Value);
-            if (_compressedSize != null) writer.Write(_compressedSize.Value);
-            if (_localHeaderOffset != null) writer.Write(_localHeaderOffset.Value);
-            if (_startDiskNumber != null) writer.Write(_startDiskNumber.Value);
+            Span<byte> extraFieldData = stackalloc byte[TotalSize];
+
+            BinaryPrimitives.WriteUInt16LittleEndian(extraFieldData[FieldLocations.Tag..], TagConstant);
+            BinaryPrimitives.WriteUInt16LittleEndian(extraFieldData[FieldLocations.Size..], _size);
+
+            if (_uncompressedSize != null)
+            {
+                BinaryPrimitives.WriteInt64LittleEndian(extraFieldData[FieldLocations.UncompressedSize..], _uncompressedSize.Value);
+            }
+
+            if (_compressedSize != null)
+            {
+                BinaryPrimitives.WriteInt64LittleEndian(extraFieldData[FieldLocations.CompressedSize..], _compressedSize.Value);
+            }
+
+            if (_localHeaderOffset != null)
+            {
+                BinaryPrimitives.WriteInt64LittleEndian(extraFieldData[FieldLocations.LocalHeaderOffset..], _localHeaderOffset.Value);
+            }
+
+            if (_startDiskNumber != null)
+            {
+                BinaryPrimitives.WriteUInt32LittleEndian(extraFieldData[FieldLocations.StartDiskNumber..], _startDiskNumber.Value);
+            }
+
+            stream.Write(extraFieldData);
         }
     }
 
-    internal struct Zip64EndOfCentralDirectoryLocator
+    internal partial struct Zip64EndOfCentralDirectoryLocator
     {
-        public const uint SignatureConstant = 0x07064B50;
-        public const int SignatureSize = sizeof(uint);
+        public static ReadOnlySpan<byte> SignatureConstantBytes => [0x50, 0x4B, 0x06, 0x07];
```

Comment:
I liked your comment above the other signature. Can you add it here too, please?:
```suggestion
        // The Zip File Format Specification references 0x07064B50, this is a big endian representation.
        // ZIP files store values in little endian, so this is reversed.
        public static ReadOnlySpan<byte> SignatureConstantBytes => [0x50, 0x4B, 0x06, 0x07];
```